identity-wallet/src/state/verified_data/reducers/mod.rs

```
@@ -0,0 +1,176 @@
+        info!("[<<<] {}", json_response);
+        let id = json_response.get("id").unwrap().as_str().unwrap();
+        let expires_at = json_response.get("expires_at").unwrap().as_str().unwrap();
+        let validation_expiration_in_secs = json_response
+            .get("validation_expiration_in_secs")
+            .unwrap()
+            .as_i64()
+            .unwrap() as i32;
+        return Ok(AppState {
+            verified_data: VerifiedData {
+                email_verification: Some(EmailVerification {
+                    email: action.email,
+                    label: action.label,
+                    verification_id: Some(id.to_string()),
+                    expires_at: Some(chrono::DateTime::parse_from_rfc3339(expires_at).unwrap().to_utc()),
+                    validation_expiration_in_secs: Some(validation_expiration_in_secs),
+                    error: None,
+                }),
+            },
+            current_user_prompt: None,
+            ..state
+        });
+    }
+    Ok(state)
+}
+
+pub async fn redeem_code(state: AppState, action: Action) -> Result<AppState, AppError> {
+    if let Some(action) = listen::<RedeemCode>(action) {
+        let session_id = state
+            .verified_data
+            .email_verification
+            .as_ref()
+            .unwrap()
+            .verification_id
+            .as_ref()
+            .unwrap();
+        let url = format!("{}/api/verify/{}", EMAIL_VERIFICATION_SERVICE_HOST, session_id);
+        let body = json!({ "code": action.code });
+        info!("[>>>] {} {}", url, body);
+        let response = reqwest::Client::new()
+            .post(url)
+            .header("X-API-KEY", EMAIL_VERIFICATION_SERVICE_API_KEY)
+            .json(&body)
+            .send()
+            .await
+            .inspect_err(|err| {
+                warn!("Failed to send verification code: {}", err);
+            })
+            .ok()
+            .unwrap();
```

This would just log a warning and then crash the app. Should we simply return the error like so?:

```rust
            .await
            .map_err(|err| AppError::Error(format!("Failed to send verification code: {err}")))?;
```